rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -698,8 +746,10 @@ public BooleanWrapper activate() {
     protected void shutdownNodeSourceServices(Client initiator) {
         logger.info("[" + name + "] Shutdown finalization");
 
-        nodeSourcePolicy.shutdown(initiator);
-        infrastructureManager.internalShutDown();
+        if (descriptor.getStatus().equals(NodeSourceStatus.NODES_DEPLOYED)) {
+            nodeSourcePolicy.shutdown(initiator);
```

Comment:
The this keyword is sometimes missing in the code. 

You can fix that automatically in IntelliJ: https://stackoverflow.com/questions/19861534/add-this-to-member-variables-methods-in-intellij-idea

Reply:
Thank you for the tip ;-)